javalib/src/main/scala/java/lang/Double.scala

```diff
@@ -441,6 +416,18 @@ object Double {
     (bits & ~scala.Long.MinValue) > PosInfinityBits
   }
 
+  /** Do `bits` correspond to a pattern for a "special" value.
+   *
+   *  Specials are zeros, infinities and NaNs.
+   */
+  @inline private[lang] def isSpecialBitPattern(bits: scala.Long): scala.Boolean = {
+    val bitsNoSign = bits & ~scala.Long.MinValue
+    if (LinkingInfo.isWebAssembly)
+      Long.unsigned_>=(bitsNoSign - 1L, PosInfinityBits - 1L) // fast long operations
```

Comment:
[note]
:+1:

- `Zero - 1`: underflow, and it's always larger than `IntinityBits - 1`
- `(Infinity or NaN) - 1`: always larger than Infinity
- Others: always smaller than Infinity